tests/integration/steps/operator.go

```diff
@@ -0,0 +1,102 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+	"github.com/avast/retry-go"
+	"github.com/kyma-project/istio/operator/tests/integration/pkg/manifestprocessor"
+	"github.com/kyma-project/istio/operator/tests/integration/testcontext"
+	v1 "k8s.io/api/apps/v1"
+	v1c "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func DeployIstioOperatorFromLocalManifest(ctx context.Context) error {
+	k8sClient, err := testcontext.GetK8sClientFromContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	resources, err := manifestprocessor.ParseFromFileWithTemplate("local-manifest.yaml", "manifests", nil)
+	if err != nil {
+		return err
+	}
+
+	return retry.Do(func() error {
+		for _, resource := range resources {
+			gvk := resource.GroupVersionKind()
+			var existingResource unstructured.Unstructured
+			existingResource.SetGroupVersionKind(gvk)
+
+			err := k8sClient.Get(ctx, client.ObjectKey{
+				Namespace: resource.GetNamespace(),
+				Name:      resource.GetName(),
+			}, &existingResource)
+
+			if err != nil {
```

Comment:
I think we are only able to safely assume that we need to create the object if this is a new resource by err being an `isNotFound` error. In all other cases we should fail it and retry.